File: app/src/main/java/social/entourage/android/map/permissions/NoLocationPermissionFragment.java
package social.entourage.android.map.permissions;

import android.app.Dialog;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.provider.Settings;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Optional;
import social.entourage.android.EntourageEvents;
import social.entourage.android.R;
import social.entourage.android.authentication.login.LoginActivity;

public class NoLocationPermissionFragment extends DialogFragment {

    public static final String TAG = "fragment_no_location_permission";

    private boolean showingGeolocationSettings = false;
    private boolean enableGeolocation = false;

    public NoLocationPermissionFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        Dialog dialog = getDialog();
        if (dialog != null) {
            Window window = dialog.getWindow();
            if (window != null) {
                window.requestFeature(Window.FEATURE_NO_TITLE);
            }
        }
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_no_location_permission, container, false);

        ButterKnife.bind(this, view);

        return view;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(final Bundle savedInstanceState) {
        return new Dialog(getActivity(), getTheme()) {
            @Override
            public void onBackPressed() {
                onBackButton();
            }
        };
    }

    @Override
    public void onStart() {
        super.onStart();

        if (getDialog() != null && getDialog().getWindow() != null) {
            getDialog().getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
            getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        if (showingGeolocationSettings) {
            onBackButton();
        }
    }

    @OnClick(R.id.no_location_back_button)
    protected void onBackButton() {
        if ( (getActivity() != null) && (getActivity() instanceof LoginActivity) ) {
            LoginActivity loginActivity = (LoginActivity)getActivity();
            loginActivity.saveGeolocationPreference(enableGeolocation);
            loginActivity.showNotificationPermissionView();
        }
        dismiss();
    }

    @OnClick(R.id.no_location_activate_button)
    protected void onActivateButton() {
        EntourageEvents.logEvent(EntourageEvents.EVENT_GEOLOCATION_ACTIVATE_04_4A);
        startActivity(new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS));
        showingGeolocationSettings = true;
        enableGeolocation = true;
    }

    @Optional
    @OnClick(R.id.no_location_ignore_button)
    protected void onIgnoreButton() {
        onBackButton();
    }
}

File: app/src/main/java/social/entourage/android/base/map/permissions/NoLocationPermissionFragment.kt
package social.entourage.android.base.map.permissions

import android.app.Dialog
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.provider.Settings
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import androidx.fragment.app.DialogFragment
import kotlinx.android.synthetic.main.fragment_no_location_permission.*
import social.entourage.android.tools.log.AnalyticsEvents
import social.entourage.android.R

class NoLocationPermissionFragment : DialogFragment() {
    private var showingGeolocationSettings = false
    private var enableGeolocation = false
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        dialog?.window?.requestFeature(Window.FEATURE_NO_TITLE)
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_no_location_permission, container, false)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return object : Dialog(requireActivity(), theme) {
            @Deprecated("Deprecated in Java")
            override fun onBackPressed() {
                onBackButton()
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        no_location_back_button?.setOnClickListener {onBackButton()}
        no_location_activate_button?.setOnClickListener {onActivateButton()}
        no_location_ignore_button?.setOnClickListener {onBackButton()} //optional button
    }

    override fun onStart() {
        super.onStart()
        dialog?.window?.let {
            it.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
            it.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        }
    }

    override fun onResume() {
        super.onResume()
        if (showingGeolocationSettings) {
            onBackButton()
        }
    }

    fun onBackButton() {
        dismiss()
    }

    fun onActivateButton() {
        AnalyticsEvents.logEvent(AnalyticsEvents.EVENT_GEOLOCATION_ACTIVATE_04_4A)
        startActivity(Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS))
        showingGeolocationSettings = true
        enableGeolocation = true
    }

    companion object {
        const val TAG = "fragment_no_location_permission"
    }
}